base/tuple.jl

```diff
@@ -576,12 +576,13 @@ function _eq(t1::Any32, t2::Any32)
 end
 
 const tuplehash_seed = UInt === UInt64 ? 0x77cfa1eef01bca90 : 0xf01bca90
-hash(::Tuple{}, h::UInt) = h + tuplehash_seed
-hash(t::Tuple, h::UInt) = hash(t[1], hash(tail(t), h))
-function hash(t::Any32, h::UInt)
-    out = h + tuplehash_seed
-    for i = length(t):-1:1
-        out = hash(t[i], out)
+function hash(tup::Tuple, h::UInt)
+    @_terminates_locally_meta
+    f = Fix2(hash, h)
+    tuph = map(f, tup)
+    out = xor(tuplehash_seed, h)
+    for g = tuph
+        out -= 3g
```

Comment:
I wonder if this should instead be `out = 3out - g`? I guess it doesn't matter, but I'm not sure.

Reply:
@StefanKarpinski any insight?

Reply:
Yeah, this change is necessary to avoid collisions, seems obvious now:

```suggestion
        out = 3*out - g
```